### PR TITLE
[plugin-video] feat: settings panel to allow download toggle

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/common/web/statics/styles/draftDefault.rtlignore.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (tamrat-bay) [inline packages/common/web/statics/styles/draftDefault.rtlignore.scss:1]: Remove change

File: packages/ricos-content/web/src/normalization/normalizeInitialState.ts [hunk 2]
```diff
@@ -146,6 +153,8 @@ const convertAnchorToLinkToUndoOneAppFix = (newEntity: RicosEntity) => {
 
 export default (content: RicosContent, config: NormalizeConfig = {}): RicosContent => {
   const { blocks, entityMap, VERSION } = processContentState(cloneDeep(content), config);
+  // console.log('new entity:', normalizeEntityMap(entityMap, config, content.VERSION || '0.0.0'));
```
Comment (tamrat-bay) [inline packages/ricos-content/web/src/normalization/normalizeInitialState.ts:156]: Remove change

File: website/docs/plugins_api/VideoPlugin.mdx [hunk 3]
```diff
@@ -47,13 +47,15 @@ import 'wix-rich-content-plugin-video/dist/styles.min.css';
 
 ## Video Config API
 
-### `disableDownload`
+### `Viewer disableDownload (deprecated!)`
+
+#### This option is no longer in available, if used the same behavior will remain. disableDownload should be set in the mediaSettings
```
Comment (shaulgo) [inline website/docs/plugins_api/VideoPlugin.mdx:52]:
```suggestion
#### disableDownload should be set in the mediaSettings of the editor.
```

File: packages/viewer/web/src/RichContentViewer.tsx [hunk 4]
```diff
@@ -116,13 +117,20 @@ class RichContentViewer extends Component<
       anchorTarget,
       relValue,
       normalize: { disableInlineImages = false, removeInvalidInlinePlugins = false },
+      config,
     } = props;
+    const { uiSettings } = config;
+    const disableVideoDownload = config[VIDEO_TYPE]?.disableDownload;
+    const disableDownload = uiSettings?.disableDownload ?? uiSettings?.disableRightClick;
```
Comment (shaulgo) [inline packages/viewer/web/src/RichContentViewer.tsx:124]:
```suggestion
    const disableDownload = config[VIDEO_TYPE]?.disableDownload || uiSettings?.disableRightClick;
```

File: packages/viewer/web/src/RichContentViewer.tsx [hunk 5]
```diff
@@ -116,13 +117,20 @@ class RichContentViewer extends Component<
       anchorTarget,
       relValue,
       normalize: { disableInlineImages = false, removeInvalidInlinePlugins = false },
+      config,
     } = props;
+    const { uiSettings } = config;
+    const disableVideoDownload = config[VIDEO_TYPE]?.disableDownload;
```
Comment (shaulgo) [inline packages/viewer/web/src/RichContentViewer.tsx:123]:
```suggestion
```

File: packages/ricos-common/web/src/RicosEngine.tsx [hunk 6]
```diff
@@ -105,9 +105,9 @@ export class RicosEngine extends Component<EngineProps> {
       onModalOpen,
       onModalClose,
     } = modalSettings;
-    const { pauseMedia, disableRightClick, fullscreenProps } = mediaSettings;
+    const { pauseMedia, disableDownload, fullscreenProps } = mediaSettings;
     const { anchorTarget, relValue } = linkSettings;
-
+    const disableRightClick = mediaSettings?.disableRightClick ?? disableDownload;
```
Comment (shaulgo) [inline packages/ricos-common/web/src/RicosEngine.tsx:110]:
```suggestion
    const disableDownload = mediaSettings?.disableRightClick || mediaSettings?.disableDownload;
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: e2e/tests/media.e2e.js [hunk 7]
```diff
@@ -348,95 +349,118 @@ describe('plugins', () => {
       cy.waitForVideoToLoad();
       cy.eyesCheckWindow(this.test.title);
     });
-  });
 
-  context('soundcloud', () => {
-    before(function() {
-      eyesOpen(this);
+    it('should toggle download option', () => {
```
Comment (shaulgo) [inline e2e/tests/media.e2e.js:353]: remove this duplicate

File: e2e/tests/media.e2e.js [hunk 8]
```diff
@@ -348,95 +349,118 @@ describe('plugins', () => {
...
+    cy.eyesCheckWindow(this.test.title);
   });
+});
 
-  context('giphy', () => {
-    before('load editor', function() {
-      eyesOpen(this);
-    });
+//TODO: fix this flaky tests
+it.skip('add a soundcloud URL', function() {
```
Comment (shaulgo) [inline e2e/tests/media.e2e.js:400]: remove this duplicate

File: packages/viewer/web/src/RichContentViewer.tsx [hunk 9]
```diff
@@ -116,13 +117,18 @@ class RichContentViewer extends Component<
       anchorTarget,
       relValue,
       normalize: { disableInlineImages = false, removeInvalidInlinePlugins = false },
+      config,
     } = props;
+    const { uiSettings } = config;
+    const disableDownload = config[VIDEO_TYPE]?.disableDownload || uiSettings?.disableRightClick;
```
Comment (shaulgo) [inline packages/viewer/web/src/RichContentViewer.tsx:123]:
```suggestion
    const disableDownload = config[VIDEO_TYPE]?.disableDownload || uiSettings?.disableRightClick || uiSettings?.disableDownload;
```

**Review by shaulgo — APPROVED** [review]

**Pull request closed without merging**